src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -181,23 +213,28 @@ public InventorySyncStatistics getStatistics() {
      *     <li>Instantiate {@code supplyChannelKeyToId} map</li>
      *     <li>Create missing supply channels if needed</li>
      * </ul>
-     * Method returns {@link CompletionStage} of {@link Map} that contains mapping of supply channels' keys to ids.
-     * It may contain {@link SyncProblemException} when fetching supply channels results in exception.
+     * Method returns {@link CompletionStage} of {@link Optional} that may contain mapping of supply channels'
+     * keys to ids. An empty optional indicates that fetching existing channels failed.
      *
      * @param drafts {@link List} containing {@link InventoryEntryDraft} objects where missing supply channels can occur
-     * @return {@link CompletionStage} of {@link Map} that contain mapping of supply channels' keys to ids, or
-     *      {@link SyncProblemException} when fetching supply channels results in exception
+     * @return {@link CompletionStage} of {@link Optional} that may contain mapping of supply channels' keys to ids.
      */
-    private CompletionStage<Map<String, String>> populateSupplyChannels(@Nonnull final List<InventoryEntryDraft>
-                                                                            drafts) {
+    private CompletionStage<Optional<Map<String, String>>> findSupplyChannels(@Nonnull final List<InventoryEntryDraft>
+                                                                                  drafts) {
         return inventoryService.fetchAllSupplyChannels()
-                .exceptionally(exception -> {
-                    syncOptions.applyErrorCallback(CTP_CHANNEL_FETCH_FAILED, exception);
-                    throw new SyncProblemException();
-                })
-                .thenApply(existingSupplyChannels -> existingSupplyChannels.stream()
-                            .collect(toMap(Channel::getKey, Channel::getId)))
-                .thenCompose(supplyChannelKeyToId -> createMissingSupplyChannels(drafts, supplyChannelKeyToId));
+            .thenApply(existingSupplyChannels -> existingSupplyChannels.stream()
+                .collect(toMap(Channel::getKey, Channel::getId)))
+            .exceptionally(exception -> {
+                syncOptions.applyErrorCallback(CTP_CHANNEL_FETCH_FAILED, exception);
+                return null;
+            })
+            .thenCompose(supplyChannelKeyToId -> {
+                if (supplyChannelKeyToId != null) {
```

Comment:
6cbdf38895da5422f9a303389df3c9dd46408a87 and then 21724e8735237ccdf6d359dc1ae8f2d15f4e5b83 (otherwise `findBugs` breaks build)